app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -153,9 +162,44 @@ class BrowserToolbarIntegration(
         toolbar.display.setUrlBackground(urlBackground)
     }
 
+    private fun setActionButtons(context: Context) {
+        if (store.state.findCustomTabOrSelectedTab(customTabId)?.isCustomTab() == false) {
+            val eraseAction = BrowserToolbar.Button(
+                imageDrawable = AppCompatResources.getDrawable(
+                    context,
+                    R.drawable.mozac_ic_delete
+                )!!,
+                contentDescription = context.getString(R.string.content_description_erase),
+                iconTintColorResource = R.color.primaryText,
+                listener = {
+                    val openedTabs = store.state.tabs.size
+                    TabCount.eraseButtonTapped.record(TabCount.EraseButtonTappedExtra(openedTabs))
+                    eraseActionListener.invoke()
+                }
+            )
+            toolbar.addNavigationAction(eraseAction)
+
+            store.flowScoped { flow ->
+                flow.ifChanged { state -> state.tabs.size }
+                    .collect {
+                        if (store.state.tabs.size > 1) {
```

Comment:
nit: If we are only interested in this check then we could make it part of the "changed" check above. Something like:

```Kotlin
flow.map { state -> state.tabs.size > 1 }
    .ifChanged()
    .collect { hasTabs ->
     // ...
     }